Explore position fixed

captbaritone opened the pull request:
I reverted https://github.com/captbaritone/webamp/pull/658 because it broke a few things:

* Automatically moving windows as you resize the browser
* The initial layout

Take a look at the last three commits to see the fixes for these issues.

Big thanks to @1j01 for the idea to try using `position: fixed` for the windows. 👏

1j01 commented:
`position: fixed` will change how the layout works tho; if you want it to still scroll with the page, you'd have to reimplement that behavior. There's no scrolling on https://deploy-preview-669--ecstatic-poincare-fe4c13.netlify.com/?useState=1 but if you add a bunch of lorem ipsum to the `body`, you can see the windows will stay fixed in the viewport versus scrolling with the page as on the current https://webamp.org/

captbaritone commented:
@1j01 Very good point. Hmm. Perhaps we return the the drawing board.

1j01 commented:
It's a shame the `position: absolute` on the container doesn't work to establish a context for `position: fixed` (I was gonna suggest/try that but it's already there, and doesn't do anything, even toggling it).

Wait, this approach may be salvageable yet!
```css
#webamp {
    transform: translate(0, 0);
}
```
From [this answer](https://stackoverflow.com/a/38796408/2624876), and also [mentioned here](https://stackoverflow.com/a/20830413/2624876)
Might want to check browser support still

(btw, test code: `
document.body.appendChild(document.createTextNode(Array(1000).fill().map(()=>"Lorem ipsum dolor sit amet")));`)

captbaritone commented:
Once again, @1j01 brilliant idea. I got it working and started testing in browsers, only to find that it does not work in Firefox.

In firefox, when you resize the browser window, the Webamp windows still end up expanding the scroll size of the window making it impossible to measure the natural size of the window.

I went ahead and pushed my, failing, changes onto this branch so others could try it out while I think about other things to try.

captbaritone commented:
I've rebased this PR on top of master, which includes a separate PR that unververted serialization. Now this PR is just the attempt at using position fixed for windows, so I've renamed it.

I don't think this approach will work, so I'll close it for now, but maybe we can use some of the ideas here to find a different solution.
